src/state.c

```diff
@@ -57,11 +57,11 @@ void backend_changed(struct xen_device *xendev, const char *node)
     if (node) {
         char *val = xs_read_be_str(xendev, node);
 
-        if (xenback->ops->backend_changed)
+        if (xenback->ops->backend_changed && val)
```

Comment:
I think we may still want to call backend_changed, even if val is NULL.  If the path is deleted, the watch will fire and val will be NULL.  We probably want to still call backend_changed in that case, and that callback should be ready for NULL.  Silently dropping a delete doesn't seem right to me.

Having said that, I think vGlass is the only user of libxenbackend now, and it doesn't defined a backend_changed callback.  So nothing is using the callback.